scripts/update_suite.rb

```diff
@@ -300,33 +302,43 @@ def parse_tests (lines)
         i = $1.to_i - 1
       end
       next if obj =~ /^\s*\/\// || obj =~ /^\s*\/\*([^*]|\*+[^*\/])*\*\/$/
-      todo << i if obj =~ /TODO|SKIP/
+      todo << i if obj =~ /(\b|\/)(TODO|SKIP)/
       tests_line[i] = obj
-      if obj =~ /RACE/ then
-        tests[i] = if obj =~ /NORACE/ then "norace" else "race" end
-      elsif obj =~ /DEADLOCK/ then
-        tests[i] = if obj =~ /NODEADLOCK/ then "nodeadlock" else "deadlock" end
-      elsif obj =~ /WARN/ then
-        tests[i] = if obj =~ /NOWARN/ then "nowarn" else "warn" end
-      elsif obj =~ /SUCCESS/ then
+      if obj =~ /(\b|\/)NOFAIL/ then
+        tests[i] = "nofail"
+      elsif obj =~ /(\b|\/)RACE/ then
+        tests[i] = "race"
+      elsif obj =~ /(\b|\/)NORACE/ then
+        tests[i] = "norace"
+      elsif obj =~ /(\b|\/)DEADLOCK/ then
+        tests[i] = "deadlock"
+      elsif obj =~ /(\b|\/)NODEADLOCK/ then
+        tests[i] = "nodeadlock"
+      elsif obj =~ /(\b|\/)WARN/ then
+        tests[i] = "warn"
+      elsif obj =~ /(\b|\/)NOWARN/ then
+        tests[i] = "nowarn"
+      elsif obj =~ /(\b|\/)SUCCESS/ then
         tests[i] = "success"
-      elsif obj =~ /FAIL/ then
+      elsif obj =~ /(\b|\/)FAIL/ then
         tests[i] = "fail"
-      elsif obj =~ /NONTERMLOOP/ then
+      elsif obj =~ /(\b|\/)NONTERMLOOP/ then
         tests[i] = "loop"
-      elsif obj =~ /NONTERMGOTO/ then
+      elsif obj =~ /(\b|\/)NONTERMGOTO/ then
         tests[i] = "goto"
-      elsif obj =~ /NONTERMFUNDEC/ then
+      elsif obj =~ /(\b|\/)NONTERMFUNDEC/ then
         tests[i] = "fundec"
-      elsif obj =~ /UNKNOWN/ then
+      elsif obj =~ /(\b|\/)UNKNOWN/ then
         tests[i] = "unknown"
-      elsif obj =~ /(assert|__goblint_check).*\(/ then
-        if obj =~ /FAIL/ then
-          tests[i] = "fail"
-        elsif obj =~ /UNKNOWN/ then
-          tests[i] = "unknown"
-        else
-          tests[i] = "assert"
+      elsif obj =~ /(\b|\/)(assert|__goblint_check).*\(/ then
+        unless obj =~ /^\s*extern\b/
```

Comment:
What's this `extern` about?

Reply:
It is about an extern definition of 'goblint_check', which would otherwise also be matched.